src/noit_check_rest.c

```diff
@@ -998,15 +999,15 @@ rest_set_check(mtev_http_rest_closure_t *restc,
       if(target) xmlFree(target);
       if(name) xmlFree(name);
       if(module) xmlFree(module);
-      if(exists) FAIL("target`name already registered");
-      if(!m) FAIL("module does not exist");
+      if(exists) FAILC(409, "target`name already registered");
```

Comment:
Is this similar to a sequence error?  I'm not sure how this condition is triggered.

Reply:
This happens if you try to instantiate two checks with the same target and name.

In circonus, all checks have names that have their circonus check_id embedded in them, so this can't happen.